File: Python/숫자_문자열과_영단어.py
def solution(s):
    strnum = ['zero','one','two','three','four','five','six','seven','eight','nine']
    answer = []
    nan = ''
    try:
        int(s)
        return int(s)
    except:
        for i in s:
            if i in list(map(lambda x: str(x), range(0,10))):
                answer.append(i)

            else:
                nan = nan + i

                if nan in strnum:
                    answer.append(strnum.index(nan))
                    nan = ''
    return int(''.join(map(str, answer)))

File: Kotlin/숫자_문자열과_영단어.kt
fun solution(s: String): Int {
    var v = s
    EngNum.values().forEach { v = v.replace(it.eng, it.num.toString()) }
    return v.toInt()
}

enum class EngNum(
    val eng: String, val num: Int
) {
    ONE("one", 1),
    TWO("two", 2),
    THREE("three", 3),
    FOUR("four", 4),
    FIVE("five", 5),
    SIX("six", 6),
    SEVEN("seven", 7),
    EIGHT("eight", 8),
    NINE("nine", 9),
    ZERO("zero", 0)
}
